### PR TITLE
Fix Klocwork issues within parameter and remote-processor

### PR DESCRIPTION
**Pull request by cc6565**
Fix Klocwork issues within parameter and remote-processor

-false positive
-referenced pointer access

It refactors iterators and make use of standard function to walk list within DomainConfiguration since KW revealed constness issues.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/249%23issuecomment-142647036%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40225653%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40299116%22%2C%20%22https%3A//github.com/01org/parameter-framework/commit/67d7ede840ced806cce0335879196691bca4bb3e%23commitcomment-13430948%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/249%23issuecomment-143189267%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/249%23issuecomment-143985372%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40684486%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/249%23issuecomment-142647036%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-23T15%3A56%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%40OznOg%20or%20%40krocard%20Please%20review.%22%2C%20%22created_at%22%3A%20%222015-09-25T11%3A13%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40dawagner%20%40miguelgaio%20%40OznOg%20%40krocard%20please%20review%20the%20last%20commit%22%2C%20%22created_at%22%3A%20%222015-09-29T08%3A29%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ca4569ceacbb3b71f4b3d38a8309746ab057d518%20parameter/DomainConfiguration.cpp%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40225653%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Use%20an%20%60std%3A%3Alist%3Cstd%3A%3Aunique_ptr%3CCAreaConfiguration%3E%3E%60%20instead%3A%20you%20won%27t%20have%20to%20implement%20anything%20in%20the%20destructor.%22%2C%20%22created_at%22%3A%20%222015-09-23T16%3A33%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.cpp%3AL53-61%22%7D%2C%20%22Pull%20ca4569ceacbb3b71f4b3d38a8309746ab057d518%20parameter/DomainConfiguration.cpp%20276%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40299116%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22walk%20through%20the%20list%20given%20in%20parameter%2C%20not%20the%20member%20one%22%2C%20%22created_at%22%3A%20%222015-09-24T09%3A13%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.cpp%3AL493-502%22%7D%2C%20%22Commit%2067d7ede840ced806cce0335879196691bca4bb3e%20parameter/DomainConfiguration.h%2013%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/commit/67d7ede840ced806cce0335879196691bca4bb3e%23commitcomment-13430948%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22tbr%22%2C%20%22created_at%22%3A%20%222015-09-25T06%3A40%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.h%3AL40%20%2867d7ede%29%22%7D%2C%20%22Pull%20e92cffbe585224f8666034e4fcecd22e03c777b8%20parameter/DomainConfiguration.cpp%2074%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/249%23discussion_r40684486%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%28%2AareaConfiguration%29.get%28%29%20-%3E%20areaConfiguration-%3Eget%28%29%22%2C%20%22created_at%22%3A%20%222015-09-29T15%3A12%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.cpp%3AL80-104%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/249#issuecomment-142647036'>General Comment</a></b>
- <a href='https://github.com/cc6565'></a> :+1:
- <a href='https://github.com/dawagner'></a> :+1: @OznOg or @krocard Please review.
- <a href='https://github.com/cc6565'></a> @dawagner @miguelgaio @OznOg @krocard please review the last commit
- [x] <a href='#crh-comment-Pull ca4569ceacbb3b71f4b3d38a8309746ab057d518 parameter/DomainConfiguration.cpp 22'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/249#discussion_r40225653'>File: parameter/DomainConfiguration.cpp:L53-61</a></b>
- <a href='https://github.com/dawagner'></a> Use an `std::list<std::unique_ptr<CAreaConfiguration>>` instead: you won't have to implement anything in the destructor.
- [x] <a href='#crh-comment-Pull ca4569ceacbb3b71f4b3d38a8309746ab057d518 parameter/DomainConfiguration.cpp 276'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/249#discussion_r40299116'>File: parameter/DomainConfiguration.cpp:L493-502</a></b>
- <a href='https://github.com/cc6565'></a> walk through the list given in parameter, not the member one
- [x] <a href='#crh-comment-Commit 67d7ede840ced806cce0335879196691bca4bb3e parameter/DomainConfiguration.h 13 40'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/commit/67d7ede840ced806cce0335879196691bca4bb3e#commitcomment-13430948'>File: parameter/DomainConfiguration.h:L40 (67d7ede)</a></b>
- <a href='https://github.com/cc6565'></a> tbr
- [ ] <a href='#crh-comment-Pull e92cffbe585224f8666034e4fcecd22e03c777b8 parameter/DomainConfiguration.cpp 74'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/249#discussion_r40684486'>File: parameter/DomainConfiguration.cpp:L80-104</a></b>
- <a href='https://github.com/krocard'></a> (*areaConfiguration).get() -> areaConfiguration->get()

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/249?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/249?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/249'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/249'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by cc6565** [comment]
:+1:

File: parameter/DomainConfiguration.cpp [hunk 1]
```diff
@@ -50,11 +53,8 @@ CDomainConfiguration::CDomainConfiguration(const string& strName) : base(strName
 
 CDomainConfiguration::~CDomainConfiguration()
 {
-    AreaConfigurationListIterator it;
-
-    for (it = _areaConfigurationList.begin(); it != _areaConfigurationList.end(); ++it) {
-
-        delete *it;
+    for (auto areaConfiguration : _areaConfigurationList) {
+        delete areaConfiguration;
```
Comment (dawagner) [inline parameter/DomainConfiguration.cpp]: Use an `std::list<std::unique_ptr<CAreaConfiguration>>` instead: you won't have to implement anything in the destructor.

File: parameter/DomainConfiguration.cpp [hunk 2]
```diff
@@ -545,11 +493,9 @@ void CDomainConfiguration::reorderAreaConfigurations(const std::list<CAreaConfig
     // Ensure elements in provided list appear first and ordered the same way in internal one
 
     // Remove all elements present in the provided list from the internal one
-    AreaConfigurationListIterator it;
-
-    for (it = areaConfigurationList.begin(); it != areaConfigurationList.end(); ++it) {
+    for (auto areaConfiguration : _areaConfigurationList) {
```
Comment (cc6565) [inline parameter/DomainConfiguration.cpp]: walk through the list given in parameter, not the member one

File: parameter/DomainConfiguration.cpp (hunk 1 above)
Comment (cc6565) [inline parameter/DomainConfiguration.cpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/DomainConfiguration.cpp (hunk 2 above)
Comment (cc6565) [inline parameter/DomainConfiguration.cpp]: :+1:<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
:+1: @OznOg or @krocard Please review.

**Comment by cc6565** [comment]
@dawagner @miguelgaio @OznOg @krocard please review the last commit

File: parameter/DomainConfiguration.cpp [hunk 3]
```diff
@@ -90,27 +80,24 @@ bool CDomainConfiguration::parseSettings(CXmlElement& xmlConfigurationSettingsEl
...
             xmlDomainImportContext.setError("Configurable Element " + strConfigurableElementPath  + " referred to by Configuration " + getPath() + " not associated to Domain");
 
             return false;
         }
-        // Ranks
-        areaConfigurationList.push_back(pAreaConfiguration);
-
         // Parse
-        if (!serializeConfigurableElementSettings(pAreaConfiguration, xmlConfigurableElementSettingsElement, xmlDomainImportContext, false)) {
+        if (!serializeConfigurableElementSettings((*areaConfiguration).get(), xmlConfigurableElementSettingsElement, xmlDomainImportContext, false)) {
```
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: (*areaConfiguration).get() -> areaConfiguration->get()

**Pull request merged**